x-pack/plugins/snapshot_restore/public/app/index.tsx

```diff
@@ -19,37 +19,45 @@ export { BASE_PATH as CLIENT_BASE_PATH } from './constants';
  */
 let DependenciesContext: React.Context<AppDependencies>;
 
-export const useAppDependencies = () => useContext<AppDependencies>(DependenciesContext);
+export const setAppDependencies = (deps: AppDependencies) => {
+  DependenciesContext = createContext<AppDependencies>(deps);
+  return DependenciesContext.Provider;
+};
+
+export const useAppDependencies = () => {
+  if (!DependenciesContext) {
+    throw new Error(`The app dependencies Context hasn't been set.
+    Use the "setAppDependencies()" method when bootstrapping the app.`);
+  }
+  return useContext<AppDependencies>(DependenciesContext);
+};
 
-const ReactApp: React.FunctionComponent<AppDependencies> = ({ core, plugins }) => {
+const getAppProviders = (deps: AppDependencies) => {
   const {
     i18n: { Context: I18nContext },
-  } = core;
+  } = deps.core;
 
-  const appDependencies: AppDependencies = {
-    core,
-    plugins,
-  };
+  // Create App dependencies context and get its provider
+  const AppDependenciesProvider = setAppDependencies(deps);
 
-  DependenciesContext = createContext<AppDependencies>(appDependencies);
-
-  return (
+  return ({ children }: { children: ReactNode }) => (
     <I18nContext>
       <HashRouter>
-        <DependenciesContext.Provider value={appDependencies}>
-          <AppStateProvider value={useReducer(reducer, initialState)}>
-            <App />
-          </AppStateProvider>
-        </DependenciesContext.Provider>
+        <AppDependenciesProvider value={deps}>
+          <AppStateProvider value={useReducer(reducer, initialState)}>{children}</AppStateProvider>
+        </AppDependenciesProvider>
       </HashRouter>
     </I18nContext>
   );
 };
 
-export const renderReact = async (
-  elem: Element,
-  core: AppCore,
-  plugins: AppPlugins
-): Promise<void> => {
-  render(<ReactApp core={core} plugins={plugins} />, elem);
+export const renderReact = async (elem: Element, core: AppCore, plugins: AppPlugins) => {
```

Comment:
Instead of providing the dependencies through props, I made it through a Provider component. This made is easier to test as I could wrap the component under test with this `Providers` Component.